packages/yew-macro/src/props/prop.rs

```diff
@@ -74,14 +76,43 @@ impl Prop {
                 "expected an expression following this equals sign",
             ));
         }
-        let value = strip_braces(input.parse::<Expr>()?)?;
+
+        let value = parse_prop_value(input)?;
         Ok(Self { label, value })
     }
 }
 
-fn strip_braces(expr: Expr) -> syn::Result<Expr> {
-    match expr {
-        Expr::Block(ExprBlock { block: Block { mut stmts, .. }, .. }) if stmts.len() == 1 => {
+fn parse_prop_value(input: &ParseBuffer) -> syn::Result<Expr> {
+    if input.peek(Brace) {
+        strip_braces(input.parse()?)
+    } else if let Some(ExprRange {
+        from: Some(from), ..
+    }) = range_expression_peek(input)
+    {
+        // If a range expression is seen, treat the left-side expression as the value
+        // and leave the right-side expression to be parsed as a base expression
+        advance_until_next_dot2(input)?;
+        Ok(*from)
+    } else {
```

Comment:
This logic here allows for the following to be valid syntax:
```rust
html! {
    <Comp prop1=some_var..props />
}
```

The `some_var` should be wrapped in braces but is accepted and used as the value for the prop, then the `..props` is parsed as a base expression, which results in the correct behaviour but it feels inconsistent. 

When the `..props` is removed the user will get the "the property value must be either a literal or enclosed.." error - so as a user I'd be confused why the braces are required now and not before, this happens even with the space between `some_var` and `..props`, 

It's a nice way to help the user get the result they probably intended but IMHO I think we should remain strict with the syntax so it stays consistent.

Reply:
Agreed! Thanks for catching that!